### ISSUE TITLE
Failed test in 32bits build

### ISSUE THREAD

**Issue by remicollet** [hprose/hprose-pecl#2]
```
 + /usr/bin/php -n run-tests.php --show-diff
 =====================================================================
 PHP         : /usr/bin/php 
 PHP_SAPI    : cli
 PHP_VERSION : 5.4.39
 ZEND_VERSION: 2.4.0
 PHP_OS      : Linux - Linux schrodingerscat.famillecollet.com 3.19.3-200.fc21.x86_64 #1 SMP Thu Mar 26 21:39:42 UTC 2015 i686
 INI actual  : /builddir/build/BUILD/php-pecl-hprose-1.0.0/NTS
 More .INIs  :  
 CWD         : /builddir/build/BUILD/php-pecl-hprose-1.0.0/NTS
 Extra dirs  : 
 VALGRIND    : Not used
 =====================================================================
 TIME START 2015-04-02 07:44:22
 =====================================================================
 TEST 1/8 [tests/bytes_io.phpt]
 PASS Test that the Hprose\BytesIO class works. [tests/bytes_io.phpt] 
 TEST 2/8 [tests/class_manager.phpt]
 PASS Test that the Hprose\ClassManager class works. [tests/class_manager.phpt] 
 TEST 3/8 [tests/formatter.phpt]
 PASS Test that the Hprose\Formatter class works. [tests/formatter.phpt] 
 TEST 4/8 [tests/raw_reader.phpt]
 ========DIFF========
 011+ d-2147483648;
 011- i-2147483648;
 013+ d2147483648;
 013- l2147483648;
 ========DONE========
 FAIL Test that the Hprose\RawReader class works. [tests/raw_reader.phpt] 
 TEST 5/8 [tests/reader.phpt]
 ========DIFF========
 011+ float(-2147483648)
 011- int(-2147483648)
 013+ float(2147483648)
 013- string(10) "2147483648"
 ========DONE========
 FAIL Test that the Hprose\Reader class works. [tests/reader.phpt] 
 TEST 6/8 [tests/result_mode.phpt]
 PASS Test that the Hprose\ResultMode class works. [tests/result_mode.phpt] 
 TEST 7/8 [tests/tags.phpt]
 PASS Test that the Hprose\Tags class works. [tests/tags.phpt] 
 TEST 8/8 [tests/writer.phpt]
 ========DIFF========
 001+ 0123456789d-2147483648;i2147483647;d2147483648;d3.1415926535898;NI-I+tfenu我s2"🆚"
 001- 0123456789i-2147483648;i2147483647;l2147483648;d3.1415926535898;NI-I+tfenu我s2"🆚"
 ========DONE========
 FAIL Test that the Hprose\Writer class works. [tests/writer.phpt] 
 =====================================================================
 TIME END 2015-04-02 07:44:22
 =====================================================================
 TEST RESULT SUMMARY
 ---------------------------------------------------------------------
 Exts skipped    :    0
 Exts tested     :   29
 ---------------------------------------------------------------------
 Number of tests :    8                 8
 Tests skipped   :    0 (  0.0%) --------
 Tests warned    :    0 (  0.0%) (  0.0%)
 Tests failed    :    3 ( 37.5%) ( 37.5%)
 Expected fail   :    0 (  0.0%) (  0.0%)
 Tests passed    :    5 ( 62.5%) ( 62.5%)
 ---------------------------------------------------------------------
 Time taken      :    0 seconds
 =====================================================================
 =====================================================================
 FAILED TEST SUMMARY
 ---------------------------------------------------------------------
 Test that the Hprose\RawReader class works. [tests/raw_reader.phpt]
 Test that the Hprose\Reader class works. [tests/reader.phpt]
 Test that the Hprose\Writer class works. [tests/writer.phpt]
 =====================================================================
```

**Comment by andot** [comment]
Thank you. I will check it.

**Comment by andot** [comment]
This problem is fixed now.

**Comment by remicollet** [comment]
I confirm, fix is OK (tested on RHEL-5 i386, PHP 5.4)
